subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/fingerprint/ConfigurationCacheFingerprintWriter.kt

```diff
@@ -264,15 +264,15 @@ class ConfigurationCacheFingerprintWriter(
     }
 
     override fun directoryChildrenObserved(directory: File, consumer: String?) {
-        if (isInputTrackingDisabled() || isExecutingTask()) {
+        if (isInputTrackingDisabled() || isExecutingWork()) {
             return
         }
         sink().captureDirectoryChildren(directory)
         reportUniqueDirectoryChildrenInput(directory, consumer)
     }
 
     override fun fileSystemEntryObserved(file: File, consumer: String?) {
-        if (isInputTrackingDisabled() || isExecutingTask()) {
+        if (isInputTrackingDisabled() || isExecutingWork()) {
```

Comment:
🎨 It seems this OR expression could be extracted to its own method to avoid multiple changes in the future.